ddtrace/span.py

```diff
@@ -324,21 +324,25 @@ def _set_str_tag(self, key, value):
         U+FFFD.
         """
         try:
-            self.meta[key] = ensure_text(value, errors="replace")
+            self._meta[key] = ensure_text(value, errors="replace")
         except Exception as e:
             if config._raise:
                 raise e
             log.warning("Failed to set text tag '%s'", key, exc_info=True)
 
     def _remove_tag(self, key):
         # type: (_TagNameType) -> None
-        if key in self.meta:
-            del self.meta[key]
+        if key in self._meta:
+            del self._meta[key]
 
     def get_tag(self, key):
         # type: (_TagNameType) -> Optional[Text]
         """Return the given tag or None if it doesn't exist."""
-        return self.meta.get(key, None)
+        return self._meta.get(key, None)
+
+    def get_tags(self):
```

Comment:
Not a fan of this new api since it returns a mutable reference to the underlying tags data structure.

eg.

```python
tags = span.get_tags()
tags["new"] = "foo"
assert span.get_tag("new") == "foo"
```

maybe a better approach is an iterator or generator over all the tags stored in meta?